src/resource-coordinator/src/ResourceCoordinator.js

```diff
@@ -0,0 +1,220 @@
+//@see https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md
+const uuid = require('uuid/v4');
+const _ = require('lodash');
+const { URL } = require('url');
+
+async function loop (body, logger, loopInterval) {
+    logger.info('loop TICK');
+    try {
+        await body();
+    } catch (e) {
+        logger.error(e, 'loop body error');
+    }
+    setTimeout(async () => {
+        loop(body, logger, loopInterval);
+    }, loopInterval);
+}
+
+class ResourceCoordinator {
+    constructor(config, logger, queueCreator, rabbitmqManagement, amqpConnection, flowsDao, driver) {
+        this._config = config;
+        this._logger = logger.child({service: 'ResourceCoordinator'});
+        this._queueCreator = queueCreator;
+        this._rabbitmqManagement = rabbitmqManagement;
+        this._channelPromise = amqpConnection.createChannel();
+        this._flowsDao = flowsDao;
+        this._driver = driver;
+    }
+
+    start() {
+        loop(this._loopBody.bind(this), this._logger, 5000);
+    }
+
+    async _loopBody() {
+        //TODO any step here blocks all the job, that's shit. Timeouts, multiple execution threads with limits for
+        //parallel jobs not to destroy backend;
+        const flows = await this._flowsDao.findAll();
+        const allApps = await this._driver.getAppList();
+        const queuesStructure = await this._getQueuesStructure();
+        await this._processState(flows, allApps, queuesStructure);
+    }
+
+    async _processState(flows, allApps, queuesStructure) {
+        const appsIndex = await this._buildJobIndex(allApps);
+        for (let flow of flows) {
+            await this._handleFlow(flow, appsIndex, queuesStructure);
+        }
+        await this._removeLostJobs(allApps, flows);
+    }
+
+    async _handleFlow(flow, appsIndex, queuesStructure) {
+        if (flow.isDeleted) {
+            this._logger.trace({name: flow.id}, 'Going to delete flow');
+
+            await this._deleteRunningAppsForFlow(flow, appsIndex);
+            await this._deleteQueuesForFlow(flow, queuesStructure);
+            await this._deleteFlowAmqpCredentials(flow);
+
+            await this._flowsDao.removeFinalizer(flow); //@todo: onFlowDeleted?
+            return;
+        }
+
+        // ensure flow infrastructure - finalizer + secret + queues
+        await this._flowsDao.ensureFinalizer(flow); //@todo: onFlowCreated?
+
+        if (await this._isRedeployRequired(flow, appsIndex)) {
+            this._logger.trace({name: flow.id}, 'Flow changed. Redeploy.');
+            await this._deleteRunningAppsForFlow(flow, appsIndex);
+            await this._deleteQueuesForFlow(flow, queuesStructure); // delete all queues? really?
```

Comment:
It's easier to remove all queues and recreate them.
Otherwise we'll have next problems
1. it's quite inconvenient to do  patch
2. old ques may contain some not processed data. If flow structure is changed we have risks that we'll feed step with some outdated messages (that was planned to put into other step)
So I don't see alternative solution.